### PR TITLE
Video player + cookie management

### PR DESCRIPTION
**Pull request by Lorezz**

### CODE REVIEW

**Comment by Lorezz** [comment]
@bfabio  ciao  fai tu review and merge ? grazie

File: source/javascripts/index.js [hunk 1]
```diff
@@ -1,34 +1,38 @@
-import React from 'react';
-import { createRoot } from 'react-dom/client';
-import 'bootstrap/dist/js/bootstrap';
+import React from "react";
```
Comment (bfabio) [inline source/javascripts/index.js:1]: The quote changes are distracting, can we leave them out?

Thinking about it, this is the `develop` branch and as you're the only one currently working on it we can rewrite the history and cleanup the un-needed/temporary commits.

I don't remember what convention this project has for quotes but if it's to use double quotes let's make a different PR for it.

File: source/javascripts/index.js [hunk 2]
```diff
@@ -71,24 +75,63 @@ for (let index = 0; index < carouselList.length; index++) {
...
   const carousel = carouselCalendarList[index];
   const carouselInstances = [];
   carouselInstances[index] = new CarouselCalendar(carousel);
 }
 
-const chartWrap = document.getElementsByClassName('chartWrap');
+// setup yt videos
+const YT_SERVICE = "youtube.com";
+const loadYouTubeVideo = (videoId) => {
+  console.log("videoId", videoId);
```
Comment (bfabio) [inline source/javascripts/index.js:89]: Let's get rid of `console.log()`

File: source/javascripts/index.js [hunk 3]
```diff
@@ -71,24 +75,63 @@ for (let index = 0; index < carouselList.length; index++) {
...
+  }
+};
+
+const overlay = document.querySelector("[data-bs-accept-overlay]");
+if (overlay) {
+  manageVideo(overlay);
+}
+
+//setup charts
+const chartWrap = document.getElementsByClassName("chartWrap");
```
Comment (bfabio) [inline source/javascripts/index.js:127]: These chart changes should be in a different PR. They also look like purely formatting changes.

File: source/javascripts/index.js [hunk 4]
```diff
@@ -98,10 +141,49 @@ if (chartWrap) {
...
+        />
       );
     } catch (error) {
-      console.log('error', error);
+      console.log("error", error);
     }
   }
 }
+
+//setup coockies management
```
Comment (bfabio) [inline source/javascripts/index.js:157]:
```suggestion
```

File: source/partials/_video-cover.html.slim [hunk 5]
```diff
@@ -9,9 +9,8 @@
 
 ruby:
   if item.video.present?
-    nocookie_url = item.video.url.gsub("youtube.com/watch?v=",
-                   "youtube-nocookie.com/embed/")
-    minheight = item.video.height < 400 ? 400 : video.height
+    nocookie_url = item.video.url.gsub("youtube.com/watch?v=","https://youtu.be/")
```
Comment (bfabio) [inline source/partials/_video-cover.html.slim:12]: We should keep the nocookie domain here

File: source/partials/_video-cover.html.slim [hunk 6]
```diff
@@ -21,22 +20,45 @@ ruby:
...
+        div class="acceptoverlay-buttons bg-dark"
+          button [
+              type="button" class="btn btn-primary" data-videoid="vid1"   data-bs-accept-from="youtube.com" data-bs-accept-button] Accetta
+          div class="form-check"
+            input [id="chk-remember" type="checkbox" data-bs-accept-remember]
+            label [for="chk-remember"] Ricorda per tutti i video
+    div
+      video [controls data-bs-video id="vid1" class="video-js" width="100%"  height=minheight  data-url=nocookie_url]
+
+  /
```
Comment (bfabio) [inline source/partials/_video-cover.html.slim:45]: Let's remove commented code

File: source/partials/links/_link_internal.html.slim [hunk 7]
```diff
@@ -3,8 +3,9 @@ ruby:
   label ||= cta_label(item)
   white_outline ||= nil
   border = white_outline ? "box-shadow: inset 0 0 0 2px white" : ""
+  page_link_path = item.anchor.present? ? page_path_with_anchor(item.link, item.anchor.present? ? "##{item.anchor}" : nil) : page_path(item.link)
```
Comment (bfabio) [inline source/partials/links/_link_internal.html.slim:6]: Is this needed for the video player or is this change unrelated?

Anyway, can't we do something like this?
```suggestion
page_link_path = item.anchor&.present? ? page_path_with_anchor(item.link, "##{item.anchor}") : page_path(item.link)

```

File: source/javascripts/index.js [hunk 8]
```diff
@@ -71,24 +75,63 @@ for (let index = 0; index < carouselList.length; index++) {
...
+  console.log("videoId", videoId);
+  const videoEl = document.getElementById(videoId);
+  const url = videoEl.dataset.url;
+  console.log("videoUrl", url);
+  const video = VideoPlayer.getOrCreateInstance(videoEl);
+  console.log("video", video);
+  video.setYouTubeVideo(url);
+};
+const manageVideo = (overlay) => {
+  const videoLoadButton = overlay.querySelector("button");
```
Comment (bfabio) [inline source/javascripts/index.js:98]: There are a lot of `querySelector()` calls and direct DOM interactions which are not great in React based app.

Take a look at https://github.com/italia/designers.italia.it/blob/main/src/components/media-player/media-player.js where AFAIR everything is opaque behind the VideoPlayer object.

Also that implementation supports subtitles and probably has a few small gotchas sorted out.

**Review by bfabio — CHANGES_REQUESTED** [review]

File: source/javascripts/index.js (hunk 1 above)
Comment (Lorezz) [reply source/javascripts/index.js:1]: this is my prettier settings, i can switch back to single quote, but for vanilla javascript is more common double quotes.

File: source/partials/_video-cover.html.slim (hunk 5 above)
Comment (Lorezz) [reply source/partials/_video-cover.html.slim:12]: We implemented the consent on youtube 3rd party cookies, so we can get the rid of old  (not working) no-cookie url

File: source/partials/links/_link_internal.html.slim (hunk 7 above)
Comment (Lorezz) [reply source/partials/links/_link_internal.html.slim]: this is needed for, internal link anchors, problably should be in another pr, but is used for the cookie consent shortcut related to video.

File: source/javascripts/index.js (hunk 1 above)
Comment (bfabio) [reply source/javascripts/index.js:1]: Let's at least put those changes in a different commit.

File: source/partials/_video-cover.html.slim (hunk 5 above)
Comment (bfabio) [reply source/partials/_video-cover.html.slim:12]: @Lorezz using youtube-nocookie.com is still slightly better, we had internal discussions about that one and came out with a best practice over the years, take a look at (unfortunately not accessible from the outside):

* https://italiandigitalteam.slack.com/archives/C3JBPM1GF/p1664355801800239?thread_ts=1664277841.817499&cid=C3JBPM1GF
* https://italiandigitalteam.slack.com/archives/CDY0TDV9P/p1645724804219149?thread_ts=1643799449.785159&cid=CDY0TDV9P

from starters, but there's most like more discussions if you look around.

File: source/javascripts/index.js (hunk 1 above)
Comment (Lorezz) [reply source/javascripts/index.js:1]: ok. i'll try

File: source/partials/_video-cover.html.slim (hunk 5 above)
Comment (Lorezz) [reply source/partials/_video-cover.html.slim:12]: ok, I'll switch back to nocookie url

File: lib/path_helpers.rb [hunk 9]
```diff
@@ -137,6 +137,10 @@ def page_path(page, locale: I18n.locale)
     end
   end
 
+  def page_path_with_anchor(page, anchor)
+      anchor ? "#{page_path(page)}#{anchor}" : page_path(page,locale)
```
Comment (bfabio) [inline lib/path_helpers.rb:141]:
```suggestion
      anchor ? "#{page_path(page)}#{anchor}" : page_path(page, locale)
```
